astropy/modeling/fitting.py

```diff
@@ -742,9 +743,14 @@ def __call__(self, model, x, y, z=None, weights=None,
         from scipy import optimize
 
         model_copy = _validate_model(model, self.supported_constraints)
-        farg = (model_copy, weights, ) + _convert_input(x, y, z)
-
-        if model_copy.fit_deriv is None or estimate_jacobian:
+        n_models = len(model_copy)
+        model_set_axis = False if n_models > 1 else model_copy.model_set_axis
+        farg = ((model_copy, weights, model_set_axis) +
+                _convert_input(x, y, z, n_models=n_models, rollaxis=False,
+                               model_set_axis=model_set_axis))
+
+        if (model_copy.fit_deriv is None or model_set_axis is False
```

Comment:
Currently I skip `_wrap_deriv` when using model sets, because I don't know how to proceed here. I have not looked in detail yet, but I didn't see a generic way to call `fit_deriv` on a set of models ?

Reply:
Yes, but broadcasting seems not so simple. I think I would to replicate the same logic as for the `evaluate` method, but it's not clear to me what's the simplest way to achieve this. I need to look more in details.